.github/workflows/ci.yml

```diff
@@ -132,8 +132,11 @@ jobs:
       BASE_ROOT_DIR: ${{ github.workspace }}
 
     steps:
-      - name: Checkout
+      - &CHECKOUT
+        name: Checkout
         uses: actions/checkout@v5
+        with:
+          ref: &CHECKOUT_REF_TMPL ${{ github.event_name == 'pull_request' && github.ref || '' }}
```

Comment:
What's the reason for running the CI on the default branch (master) when we push to any branch on the repo?

I think it makes sense to set the ref to the PR or the branch?